src/Psl/Regex/matches.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Regex;
+
+use Psl\Internal;
+use Psl\Str;
+
+/**
+ * Determine if $subject matches the given $pattern.
+ *
+ * @param non-empty-string  $pattern    The pattern to match against.
+ *
+ * @throws Exception\InvalidPatternException If $pattern is invalid.
+ *
+ * @psalm-pure
+ */
+function matches(string $subject, string $pattern, int $offset = 0): bool
```

Comment:
Maybe we could introduce a separate function to get the matches from a pattern?

Reply:
Indeed, most of the time you want to know the matching patterns as well.

We could use 2 functions like:

- `matches(string $subject, string $pattern, int $offset=0) : bool`
- `matching(string $subject, string $pattern, TypeInterface $shape, int $offset=0) : dict`

That way, the shape validates the result and makes sure it is type-safe?

Reply:
planning on it, just not the goal of this PR, not sure if i would use `TypeInterface` to force the type, or find another solution, but for now, this PR adds only basic things